src/doc/src/reference/build-scripts.md

```diff
@@ -74,16 +74,27 @@ directory specified in the [`OUT_DIR` environment variable][build-env]. Scripts
 should not modify any files outside of that directory.
 
 Build scripts communicate with Cargo by printing to stdout. Cargo will
-interpret each line that starts with `cargo:` as an instruction that will
+interpret each line that starts with `cargo::` as an instruction that will
 influence compilation of the package. All other lines are ignored.
 
-> Note: The order of `cargo:` instructions printed by the build script *may*
+After the merge of [#12201], various instruction formats are now available, including:
```

Comment:
I am not sure does this doc clear enough or not. Do you have any suggestions about how to explain it to users?

Reply:
Personally I think we don't need to mention the old syntax that much, given it is deprecated (yep it IS not WILL be deprecated). Probably we can just add a note like

> **Note:** The old invocation prefix `cargo:` (one colon only) is deprecated and planned to be removed. To migrate, use two-colons prefix `cargo::` instead. If you were using `cargo:KEY=VALUE` for arbitrary `links` manifest key-value pair, use `cargo::metadata=KEY=VALUE`.


And maybe, I said maybe, we can remove it in Edition 2024, forcing the transition.

Reply:
> And maybe, I said maybe, we can stop supporting it for package in Edition 2024, forcing the transition.

Unfortunately, a package may delegate to a dependency from a prior edition to print output.

We could recursively check all `build-dependencies` editions?

Reply:
We should probably mention in the comment what Rust version is needed for the new syntax to help people straddling this with their MSRV

Reply:
Updated.